spark/src/main/scala/geotrellis/spark/tiling/MapKeyTransform.scala

```diff
@@ -95,12 +103,16 @@ class MapKeyTransform(val extent: Extent, val layoutCols: Int, val layoutRows: I
     (tcol.floor.toInt, trow.floor.toInt)
   }
 
-  def apply[K: SpatialComponent](key: K): Extent = {
-    apply(key.getComponent[SpatialKey])
-  }
+  def keyLikeToExtent[K: SpatialComponent](key: K): Extent = apply(key)
```

Comment:
Yeah, I think `SpatialKey => Extent` signals its intent better. If the user knows what the purpose of `SpatialComponent` is, then should be able to make the `getComponent` call themselves (if say their key type was `SpaceTimeKey`).